examples/tensorflow/real-time-object-detection-rpi4/real_time_detection_with_labels_on_raspberry4.py

```diff
@@ -0,0 +1,154 @@
+
+import argparse
+import sys
+import time
+
+import cv2
+import mediapipe as mp
+
+from mediapipe.tasks import python
+from mediapipe.tasks.python import vision
+
+from utils import visualize
+from picamera2 import Picamera2
+
+# Global variables to calculate FPS
+COUNTER, FPS = 0, 0
+START_TIME = time.time()
+picam2 = Picamera2()
+picam2.preview_configuration.main.size = (640,480)
+picam2.preview_configuration.main.format = "RGB888"
+picam2.preview_configuration.align()
+picam2.configure("preview")
+picam2.start()
+
+def run(model: str, max_results: int, score_threshold: float, 
+        camera_id: int, width: int, height: int) -> None:
+  """Continuously run inference on images acquired from the camera.
+
+  Args:
+    model: Name of the TFLite object detection model.
+    max_results: Max number of detection results.
+    score_threshold: The score threshold of detection results.
+    camera_id: The camera id to be passed to OpenCV.
+    width: The width of the frame captured from the camera.
+    height: The height of the frame captured from the camera.
+  """
+
+
+
+  # Visualization parameters
+  row_size = 50  # pixels
+  left_margin = 24  # pixels
+  text_color = (0, 0, 0)  # black
+  font_size = 1
+  font_thickness = 1
+  fps_avg_frame_count = 10
+
+  detection_frame = None
+  detection_result_list = []
+
+  def save_result(result: vision.ObjectDetectorResult, unused_output_image: mp.Image, timestamp_ms: int):
+      global FPS, COUNTER, START_TIME
+
+      # Calculate the FPS
+      if COUNTER % fps_avg_frame_count == 0:
+          FPS = fps_avg_frame_count / (time.time() - START_TIME)
+          START_TIME = time.time()
+
+      detection_result_list.append(result)
+      COUNTER += 1
+
+  # Initialize the object detection model
+  base_options = python.BaseOptions(model_asset_path=model)
+  options = vision.ObjectDetectorOptions(base_options=base_options,
+                                         running_mode=vision.RunningMode.LIVE_STREAM,
+                                         max_results=max_results, score_threshold=score_threshold,
+                                         result_callback=save_result)
+  detector = vision.ObjectDetector.create_from_options(options)
+
+
+  # Continuously capture images from the camera and run inference
+  while True:
+    im= picam2.capture_array()  
+#    success, image = cap.read()
+    image=cv2.resize(im,(640,480))
+    image = cv2.flip(image, -1)
```

Comment:
Could we ask for a flipped image from the camera system in the first place? That's basically free so would be more efficient, though we obviously have to be sure it doesn't break the demo in some way.